lib/esConnector.js

```diff
@@ -1266,6 +1276,309 @@ ESConnector.prototype.updateOrCreate = function updateOrCreate(modelName, data,
+    dataBulk.push(action, clonedData);
+
+    return dataBulk;
+  }, [ /* dataBulk */ ]);
+
+  return self
+    .bulk(modelName, bulkParams)
+    .then(
+      function(res){
+        log('ESConnector.prototype.bulkCreate', 'successful executed in: ' + res.took + 'ms');
+        
+        var result = res.items.map(function(item){
+          return item[actionName]._id;
+        });
+
+        if (typeof done === 'function') {
+          done(null, result);
+        } else {
+          return Promise.resolve(result);
+        }
+      },
+      function(err){
+        console.trace(err.message);
+        if (typeof err !== 'undefined' && err !== null) {
+          if (typeof done === 'function') {
+            done(err, null);
+          } else {
+            return Promise.reject(err);
+          }
+        }
+      }
+    );
+};
+
+
+/**
+ * ES bulk update
+ * @param modelName {String} - required, model that is associated with this connector
+ * @param dataList {Array} - required, list of documents/models containing its ID and the changed
+ * @param params {Object} - passed, through to the bulk API
+ * @param done {Function} - required, allows async execution
+ * @returns {Promise} OR {undefined} if callback (done) is provided
+ * @value: {Number} - return the amount of affected items
+ * @throws {Error} - if argument rules are violated
+ *
+ * NOTE: if params has a field called `body`, it will be ignored. Instead dataList is used
+ */
+ESConnector.prototype.bulkUpdate = function bulkUpdate(modelName, dataList, params, done){
+  var self = this;
+  var functionName = 'bulkUpdate';
+  if (self.debug) {
+    log('ESConnector.prototype.' + functionName, modelName, dataList, params);
+  }
+
+  if (Array.isArray(dataList) === false) {
+    var typeError = new Error('invalid type for dataList');
+    if (typeof done === 'function') {
+        return done(typeError);
+    } else {
+        return Promise.reject(typeError);
+    }
+  }
+
+  if (typeof params === 'function') {
+    done = params;
+    params = {};
+  } else if (typeof params !== 'object' || params === null) {
+    params = {};
+  }
+
+  var idName = self.idName(modelName);
+  var actionName = 'update';
+  var defaultParams = self.addDefaults(modelName, functionName);
+  var bulkParams = _.assign({}, defaultParams, params);
+
+  bulkParams.body = dataList.reduce(function(dataBulk, data){
+    var idValue = self.getIdValue(modelName, data);
+
+    var clonedData = _.assign({}, data);
+
+    delete clonedData[idName];
+    var action = {};
+    action[actionName] = { _id: idValue };
+    dataBulk.push(action, { doc: clonedData });
+
+    return dataBulk;
+  }, [ /* dataBulk */ ]);
+
+  return self
+    .bulk(modelName, bulkParams)
+    .then(
+      function(res){
+        log('ESConnector.prototype.bulkUpdate', 'successful executed in: ' + res.took + 'ms');
+
+        var affectedAmount = res.items.reduce(function(amount, item){
+          if(item[actionName].result === 'updated'){
```

Comment:
Can you please point me to one of the ES doc versions where sample response has the property `result`?